server/lib/genome/importers/api_importers/civic/importer.rb

```diff
@@ -9,14 +9,87 @@ class Importer < Genome::Importers::Base
 
           def initialize
             @source_db_name = 'CIViC'
+            @api_client = ApiClient.new
           end
 
           def create_claims
+            create_drug_claims
+            create_gene_claims
             create_interaction_claims
           end
 
           private
 
+          # this should be rare -- maybe impossible under the latest schema changes?
+          def importable_drug?(drug)
+            drug.name.upcase != 'N/A' && !drug.name.include?(';')
+          end
+
+          def importable_evidence_item?(ei)
+              [
+                ei.evidence_direction != 'SUPPORTS',
+                ei.evidence_level == 'E',
+                ei.evidence_rating.present? && ei.evidence_rating <= 2
+              ].any?
+          end
+
+          def create_drug_claims
+            @api_client.enumerate_drugs.each do |drug|
+              next if !importable_drug?(drug)
+
+              dc = create_drug_claim(drug.name.upcase, DrugNomenclature::PRIMARY_NAME)
+              create_drug_claim_alias(dc, "civic.tid:#{drug.id}", DrugNomenclature::CIVIC_TID)
+              create_drug_claim_alias(dc, "ncit:#{drug.ncit_id}", DrugNomenclature::NCIT_ID) if drug.ncit_id
+            end
+          end
+
+          def create_gene_claims
+            @api_client.enumerate_genes.each do |gene|
+              gc = create_gene_claim(gene.name.upcase, GeneNomenclature::SYMBOL)
+              create_gene_claim_alias(gc, "civic.gid:#{gene.id}", GeneNomenclature::CIVIC_GID)
+              create_gene_claim_alias(gc, "ncbi.gene:#{gene.entrez_id}", GeneNomenclature::NCBI_ID) if gene.entrez_id
+              create_gene_claim_alias(gc, gene.full_name, GeneNomenclature::NAME) if gene.full_name
+              gene.feature_aliases.uniq.reject { |n| n == gene.name }.each do |gene_alias|
+                create_gene_claim_alias(gc, gene_alias, GeneNomenclature::SYMBOL)
+              end
+            end
+          end
+
+          def create_entries_for_evidence_item(ei, dc, gc)
+            ic = create_interaction_claim(gc, dc)
+            if ei.source.citation_id.present? && ei.source.source_type == 'PubMed'
+              create_interaction_claim_publication(ic, ei.source.citation_id)
+            end
+            create_interaction_claim_link(ic, "EID#{ei.id}", "https://civicdb.org/evidence/#{ei.id}")
+          end
+
+          # current policy is to skip items with poor rating/level, anything that doesn't support the claim,
+          # or any item with >1 genes (e.g. a fusion)
+          def create_interaction_claims
+            @api_client.enumerate_evidence_items.each do |ei|
+              next if !importable_evidence_item?(ei)
+
+              # retain molecular profiles consisting of multiple variations on the same gene
+              gene_names = ei.molecular_profile.variants.map { |v| v.feature.feature_instance.name.upcase }.uniq
```

Comment:
A `feature` is 1:1 with a `featureInstance`. Essentially a `feature` exists in the schema where `gene` used to be, and then `featureInstance` is where any fields specific to the given feature type (ie: entrez id for a Gene, ncit id for a Factor, etc).

This a longwinded way of saying that any variants that share a feature_id share the same gene (assuming the feature is in fact a gene), so if you wanted to just make sure all the variants had the same feature id, that might be more straightfoward than name matching.

If you want to exclude non-gene variants, you could pull back the `__typename` of the feature instance, make sure its `Gene`. If that's true and all the ids match, you shouldn't need to do any further comparisons.